base/intfuncs.jl

```diff
@@ -639,10 +639,25 @@ ndigits(x::Integer; base::Integer=10, pad::Integer=1) = max(pad, ndigits0z(x, ba
 
 ## integer to string functions ##
 
-function bin(x::Unsigned, pad::Int, neg::Bool)
-    m = 8 * sizeof(x) - leading_zeros(x)
-    n = neg + max(pad, m)
-    a = StringVector(n)
+const PRINT_INT_SCRATCH_LENGTH = 256
+const PRINT_INT_SCRATCH = Vector{UInt8}[StringVector(PRINT_INT_SCRATCH_LENGTH)]
```

Comment:
I would try to avoid a global cache like this, since it will lead to data races when used from multiple threads simultaneously. Instead, this should probably be passed around from function to function.

Reply:
You're right that initialization could be done better. But as I said above, I need some help with that because show() and print() are required to work before __init__() has been called (please correct me if I'm wrong).

Not entirely sure what you mean by data races, as the vector is (meant to be) indexed by threadid(), hence no data races are possible.

Reply:
> as the vector is (meant to be) indexed by threadid()

Ah, I didn't read carefully enough.

I am still not convinced relying on global state here is the prefered solution. Why not just let the caller pass in a preallocated vector that is used instead, if something is performance critical?

Reply:
What caller do you have in mind? To me, user code calling `show()`/`print()` shouldn't be responsible for providing a scratch buffer for integer serialization :-)

Or are you suggesting `print_int()` to take buffer as an argument and pass it down to bin/oct/dec/hex? This sounds reasonable, but wouldn't ultimately solve the need for thread-local scratch-space, only this time it will be requested by `show()`/`print()`/etc